src/contracts/GPv2SimpleAuthentication.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: LGPL-3.0-or-newer
+pragma solidity ^0.6.12;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/EnumerableSet.sol";
+import "./interfaces/GPv2Authentication.sol";
+
+/// @title Gnosis Protocol v2 Access Control Contract
+/// @author Gnosis Developers
+contract GPv2SimpleAuthentication is Ownable, GPv2Authentication {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    EnumerableSet.AddressSet private solvers;
+
+    function addSolver(address solver) public onlyOwner {
+        solvers.add(solver);
+    }
+
+    function removeSolver(address solver) public onlyOwner {
+        solvers.remove(solver);
+    }
+
+    function isSolver(address prospectiveSolver)
+        public
```

Comment:
Not super important anymore (since the interface is defined as `external`) but all these methods should probably be `external`.

Reply:
Although I will have to create a test interface then in order to test this I suppose.

Reply:
Why? You can call `external` functions, only `internal` ones require a test interface to make them public.

Reply:
All good. external is it.